Factory.sol

```diff
@@ -0,0 +1,71 @@
+pragma solidity ^0.5.2;
+import {positionContract} from "./positionContract.sol";
+import {CErc20, ComptrollerInterface, CToken} from "./CERC20.sol";
+import {CEther} from "./CEther.sol";
+import {UniswapExchangeInterface} from "./uniswap.sol";
+import {ERC20Interface} from "./ERC20.sol";
+
+contract Factory{
+    mapping (uint => address) public tokenAddresses;
+    mapping (uint => address) public ctokenAddresses;
+    mapping (uint => address) public tokenExchangeAddresses;
+    
+    // userAddr => shortRepAddr
+    mapping (address => address) public REP;
+    mapping (address => address) public ZRX;
+    
+    constructor() public {
+    // DAI    
+    tokenAddresses[0] = 0x5592EC0cfb4dbc12D3aB100b257153436a1f0FEa;
+    // BAT
+    // no liquidity on this. 
+    tokenAddresses[1] = 0xbF7A7169562078c96f0eC1A8aFD6aE50f12e5A99;
+    // ZRX
+    tokenAddresses[2] = 0xddea378A6dDC8AfeC82C36E9b0078826bf9e68B6;
+    // REP
+    tokenAddresses[3] = 0x6e894660985207feb7cf89Faf048998c71E8EE89;
+        
+    ctokenAddresses[0] = 0x2ACC448d73e8D53076731fEA2EF3fc38214d0A7d;
```

Comment:
Do these ever change? You can just specify these as constants instead of setting storage in ctor. It will be cleaner and costs less gas to deploy.

Reply:
Oh, I guess solidity probably doesn't allow map literals 😱